### PR TITLE
Golden path lab scripts

### PR DESCRIPTION
**Pull request by msl8r**
### Change description ###
- Added guide for creating and deleting flux config for the java golden path tutorial 
- Updated the add image polices script to create the component folder if it doesn't exists
- Added create and clean up script for flux config

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: labs/java/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+The above script will do the following:
+- Create the directory that the new config will be created in `apps/labs/labs-'yourGitHubUsername'`
+- Create the imagePolicy and imageRepository config
+- Create the HelmRelease resource to deploy your app 
+- Add the image-policy and image-repo file references to the labs automation kustomization file so flux can automate image updates
+- Add a reference to the HelmRelease file the sbox kustomization cluster so the resources are 
+
+Once the script has been exectued, you will need to add and commit the new config and push it to GitHub. 
+- Run `git status` to check which files have been create and modified.
+- Use `git add` to stage the newly created files for commit
```
Comment (timja) [inline labs/java/README.md:21]: add either . or a path if there's a common path
```suggestion
- Use `git add .` to stage the newly created files for commit
```

File: labs/java/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+- Run `git diff` to view the changes made to the existing kustomization files. You should see references to the newly created files for your lab app.
```
Comment (timja) [inline labs/java/README.md:22]: diff won't work after adding

```suggestion
- Run `git diff --staged` to view the changes made to the existing kustomization files. You should see references to the newly created files for your lab app.
```

File: labs/java/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+- Once you're happy with the changes made, commit and push them and create a PR.
+
+More information on flux v2, its components and custom resources can be found on [flux docs site](https://fluxcd.io/docs/concepts/).
+
+## Deleting the flux config for your Java lab application
+
+Once you're finished with your lab app you can clean up the flux configuration but running the [`clean-up-lab-flux-config.sh`](./create-lab-flux-config.sh) script.
```
Comment (timja) [inline labs/java/README.md:29]:
```suggestion
Once you're finished with your lab app you can clean up the flux configuration by running the [`clean-up-lab-flux-config.sh`](./create-lab-flux-config.sh) script. 
```

File: labs/create-lab-flux-config.sh [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+# Create HR for lab
+(
+cat <<EOF
+---
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: labs-${COMPONENT}
+  namespace: labs
```
Comment (timja) [inline labs/create-lab-flux-config.sh:52]: why not have this script as generic for anyone adding a helm release?

File: labs/create-lab-flux-config.sh [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    spec:
+      chart: ./stable/labs-${COMPONENT}
+      sourceRef:
+        kind: GitRepository
+        name: hmcts-charts
+        namespace: flux-system
+      interval: 1m
+  values:
+    ${APPLICATION}:
+      image: hmctssandbox.azurecr.io/labs/${COMPONENT}:latest #{"\$imagepolicy": "flux-system:labs-${COMPONENT}"}
```
Comment (timja) [inline labs/create-lab-flux-config.sh:65]: should have a space so lint doesn't fail I think
```suggestion
      image: hmctssandbox.azurecr.io/labs/${COMPONENT}:latest # {"\$imagepolicy": "flux-system:labs-${COMPONENT}"}
```

File: labs/create-lab-flux-config.sh [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+      disableTraefikTls: true
+    global:
+      environment: sandbox
+      tenantId: "531ff96d-0ae9-462a-8d2d-bec7c0b42082"
+EOF
+) > "${COMPONENT_DIR}/${PRODUCT}-${COMPONENT}.yaml"
+
+if [[ ${APPLICATION} == "nodejs" ]]; then
+  export BACKEND="http://${PRODUCT}-${COMPONENT}-sandbox.service.core-compute-sandbox.internal"
+  yq eval  -i '(.spec.values.nodejs.environment.RECIPE_BACKEND_URL) = env(BACKEND) ' ${COMPONENT_DIR}/${PRODUCT}-${COMPONENT}.yaml
```
Comment (timja) [inline labs/create-lab-flux-config.sh:76]:
```suggestion
  yq eval -i '(.spec.values.nodejs.environment.RECIPE_BACKEND_URL) = env(BACKEND) ' ${COMPONENT_DIR}/${PRODUCT}-${COMPONENT}.yaml 
```

File: labs/README.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
+# Golden Path - Creating and deleting Flux config for an application
+
+[Golden Path tutorial - Java](https://backstage.platform.hmcts.net/catalog/default/component/golden-path-java) - VPN needed.
+
+## Creating the flux config for your lab application 
+
+Inside the labs directory run the [`create-lab-flux-config.sh`](./create-lab-flux-config.sh) script to create the flux config needed for your lab application. Use the component name you used when creating your app with backstage, which should be your GitHub username and specify the type of application you are deploying.
```
Comment (timja) [inline labs/README.md:7]: scripts should never depend on your current working directory, please adding something like https://stackoverflow.com/questions/3349105/how-can-i-set-the-current-working-directory-to-the-directory-of-the-script-in-ba

File: labs/README.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+<details>
+<summary>Command to create the flux configuration for a Java app</summary>
```
Comment (timja) [inline labs/README.md:10]: just have them expanded, people often miss these and for something so short it's fine

File: labs/README.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+<details>
+<summary>Command to create the flux configuration for a Node.js app</summary>
+```shell
+./create-lab-flux-config.sh yourGitHubUsername nodejs
+```
+</details>
+
+The above script will do the following:
+- Create the directory that the new config will be created in `apps/labs/labs-'yourGitHubUsername'`
+- Create the imagePolicy and imageRepository config
```
Comment (timja) [inline labs/README.md:25]: aren't these capitals?
```suggestion
- Create the ImagePolicy and ImageRepository config
```

File: bin/v2/add-image-policies.sh [hunk 10]
```diff
@@ -6,8 +6,11 @@ PRODUCT=$2
 COMPONENT=$3
 REGISTRY=$4
 ACR=${REGISTRY:-hmctspublic}
-COMPONENT_DIR="apps/${NAMESPACE}/${PRODUCT}-${COMPONENT}"
+APPS_DIR="../../apps/"
+COMPONENT_DIR="${APPS_DIR}/${NAMESPACE}/${PRODUCT}-${COMPONENT}"
 
+cd "$(dirname "$0")"
+pwd
```
Comment (timja) [inline bin/v2/add-image-policies.sh]: probably don't need to log this? assume it was just debug

File: labs/create-lab-flux-config.sh [hunk 11]
```diff
@@ -28,59 +29,59 @@ if [ -d "${COMPONENT_DIR}" ]; then
...
-    ${APPLICATION}:
-      image: hmctssandbox.azurecr.io/labs/${COMPONENT}:latest # {"\$imagepolicy": "flux-system:labs-${COMPONENT}"}
-      ingressHost: ${INGRESS_HOST}
-      disableTraefikTls: true
-    global:
-      environment: sandbox
-      tenantId: "531ff96d-0ae9-462a-8d2d-bec7c0b42082"
-EOF
-) > "${COMPONENT_DIR}/${PRODUCT}-${COMPONENT}.yaml"
+# if [[ ${APPLICATION} == java ]]; then
```
Comment (timja) [inline labs/create-lab-flux-config.sh]: why commented out?

File: bin/v2/add-image-policies.sh (hunk 10 above)
Comment (msl8r) [reply bin/v2/add-image-policies.sh]: yeah it was i've removed it 👍🏼

File: labs/create-lab-flux-config.sh (hunk 11 above)
Comment (msl8r) [reply labs/create-lab-flux-config.sh]: Moved it out to add-helm-release.sh

File: labs/README.md (hunk 7 above)
Comment (timja) [reply labs/README.md:7]: not pushed? scripts should be linked from the root of the repo, so probably:

```
./labs/create-lab-flux-config.sh
```

Comment (msl8r) [reply labs/README.md]: ah sorry thought you meant something else, updated 👍🏼

File: apps/probate/automation/kustomization.yaml [hunk 12]
```diff
@@ -21,3 +21,5 @@ resources:
   - ../probate-caveats-fe/image-repo.yaml
   - ../probate-caveats-fe/image-policy.yaml
   - ../probate-caveats-fe/demo-image-policy.yaml
+  - ../probate-myapp/image-repo.yaml
+  - ../probate-myapp/image-policy.yaml
```
Comment (timja) [inline apps/probate/automation/kustomization.yaml:25]: why are you adding 2 here?

File: bin/v2/add-helm-release.sh [hunk 13]
```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+set -ex
+
+NAMESPACE=${1}
+PRODUCT=${2}
+COMPONENT=${3}
+ACR=${4}
+NAMESPACE_DIR="../../apps/${NAMESPACE}"
+COMPONENT_DIR="${NAMESPACE_DIR}/${PRODUCT}-${COMPONENT}"
+APPLICATION=${5}
```
Comment (timja) [inline bin/v2/add-helm-release.sh:10]: APPLICATION_LANGUAGE or LANGUAGE I think would be a better name?

File: apps/probate/automation/kustomization.yaml (hunk 12 above)
Comment (msl8r) [reply apps/probate/automation/kustomization.yaml]: removed

**Review by zc-hmcts — APPROVED** [review]

**Pull request merged**